src/cascadia/TerminalSettingsEditor/ProfileViewModel.cpp

```diff
@@ -328,24 +409,89 @@ namespace winrt::Microsoft::Terminal::Settings::Editor::implementation
         }
     }
 
-    bool ProfileViewModel::HideIcon()
+    winrt::hstring ProfileViewModel::LocalizedIcon() const
     {
-        return Icon() == HideIconValue;
-    }
-    void ProfileViewModel::HideIcon(const bool hide)
-    {
-        if (hide)
+        if (_currentIconType && unbox_value<IconType>(_currentIconType.as<Editor::EnumEntry>().EnumValue()) == IconType::None)
         {
-            // Stash the current value of Icon. If the user
-            // checks and un-checks the "Hide Icon" checkbox, we want
-            // the path that we display in the text box to remain unchanged.
-            _lastIcon = Icon();
-            Icon(HideIconValue);
+            return RS_(L"Profile_IconTypeNone");
         }
-        else
+        return Icon();
+    }
+
+    void ProfileViewModel::CurrentIconType(const Windows::Foundation::IInspectable& value)
+    {
+        if (_currentIconType != value)
         {
-            Icon(_lastIcon);
+            // Switching from...
+            if (_currentIconType && unbox_value<IconType>(_currentIconType.as<Editor::EnumEntry>().EnumValue()) == IconType::Image)
+            {
+                // Stash the current value of Icon. If the user
+                // switches out of then back to IconType::Image, we want
+                // the path that we display in the text box to remain unchanged.
+                _lastIconPath = Icon();
+            }
+
+            // Set the member here instead of after setting Icon() below!
+            // We have an Icon property changed handler defined for when we discard changes.
+            // Inadvertently, that means that we call this setter again.
+            // Setting the member here means that we early exit at the beginning of the function
+            //  because _currentIconType == value.
+            _currentIconType = value;
+
+            // Switched to...
+            switch (unbox_value<IconType>(value.as<Editor::EnumEntry>().EnumValue()))
+            {
+            case IconType::None:
+            {
+                Icon(HideIconValue);

```

Comment:
do we need to notify changes here?

Reply:
or, do we only notify below (R456) because we want the text box to update

Reply:
Oh! My bad. The `_NotifyChanges(L"Icon");` on R456 is redundant. `Icon()` automatically notifies changes. Removing R456. Must've been left over from an earlier iteration 😅